src/main/java/com/superdevs/HealthOMeter/entity/Contact.java

```diff
@@ -0,0 +1,232 @@
+package com.superdevs.HealthOMeter.entity;
+
+import javax.validation.constraints.NotBlank;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Entity(name = "contacts")
+public class Contact {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private long id;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "contacts")
+    private Set<BioMetricData> bioMetricData;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "contacts")
+    private Set<BioMetricHistoricalData> bioMetricHistoricalData;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "contacts")
+    private Set<BMIData> bmiData;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "contacts")
+    private Set<WHRData> whrData;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "contacts")
+    private Set<RFMData> rfmData;
+
+    @ManyToMany
+    @JoinTable(
+            name = "historicalcalculatorsresults_contact",
+            joinColumns = @JoinColumn(name = "contact_id"),
+            inverseJoinColumns = @JoinColumn(name = "historicalcalculatorsresults_id")    )
+    private Set<HistoricalCalculatorsResults> historicalCalculatorsResults;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "medicaldata_id", referencedColumnName = "id")
+    private MedicalData medicalData;
+
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "contacts")
+    private Set<HistoricalMedicalData> historicalMedicalData;
+
+    @Column(name = "first_name")
+    @NotBlank(message = "First name must not be null or empty!")
+    private String firstName;
+
+    @Column(name = "last_name")
+    private String lastName;
+
+    @Column(name = "email")
+    @NotBlank(message = "Email must not be null or empty!")
+    private String email;
+
+    @Column(name = "nick_name")
+    @NotBlank(message = "Nick name must not be null or empty!")
+    private String nickName;
+
+    @Column(name = "country")
+    private String country;
+
+    @Column(name = "city")
+    private String city;
+
+    @Column(name = "created")
+    private LocalDateTime created;
+
+    @Column(name = "updated")
+    private LocalDateTime updated;
+
+    public Contact() {
+    }
+
+    public Contact(long id, String firstName, String email, String nickName) {
```

Comment:
zostanmy przy konstruktorach empty i typu all-in. Jak bedzie tzreba jakiegos innego to dodamy podczas implementacji

Reply:
Changed.